### ISSUE TITLE
Migrate to the Common Project System

### ISSUE THREAD

**Issue by sharwell** [EWSoftware/SHFB#90]
The Common Project System is the de-facto standard way to add support for new project types in recent versions of Visual Studio. With #83 moving us exclusively to Visual Studio 2013+, it should¹ be the cleanest and easiest way to provide a great project system.

¹ Change this to "will" once we test it out! :wink:

**Comment by EWSoftware** [comment]
Just to clarify, by "Common Project System", are you referring to your MPFProj10 from NuGet that's an improved version of what's currently in the project or is this something else?  If something else, can you provide some references?  I can't find anything.

**Comment by sharwell** [comment]
Documentation is virtually non-existent right now, but it's coming.

**Comment by EWSoftware** [comment]
So, not MPFProjec10?

**Comment by sharwell** [comment]
That is correct. CPS was originally created for the new C++ project system in Visual Studio 2010, and over time expanded to support a wider range of scenarios internal to Microsoft. Starting with Visual Studio 2015, they are publicly documenting the system. However, an older version of the assemblies is still included in Visual Studio 2013 (stable but not documented), so I plan to craft a solution for SHFB that works in both environments.

**Comment by sharwell** [comment]
The documentation for the new project system is being added here:
https://github.com/Microsoft/VSProjectSystem

I'll give a better overview of the work involved with migrating SHFB to use this after a more extensive review and experimentation.

**Comment by EWSoftware** [comment]
Looks interesting though not yet finished.  I'm doubtful about using it yet until the API stabilizes simply because I don't want to maintain separate versions and/or installers for each version of Visual Studio until it does.
